kyuubi-server/src/main/scala/org/apache/kyuubi/engine/JpsApplicationOperation.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine
+
+import java.nio.file.Paths
+
+import scala.sys.process._
+
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.ApplicationOperation.NOT_FOUND
+
+class JpsApplicationOperation extends ApplicationOperation {
+
+  private val runner = {
+    val jps = sys.env.get("JAVA_HOME").orElse(sys.props.get("java.home"))
+      .map(Paths.get(_, "bin", "jps").toString)
+      .getOrElse("jps")
+    try {
+      jps.!!
+      jps
+    } catch {
+      case _: Throwable => null
+    }
+  }
+
+  override def initialize(conf: KyuubiConf): Unit = {}
+
+  override def isSupported(clusterManager: Option[String]): Boolean = {
+    runner != null && clusterManager.isEmpty || clusterManager.get == "local"
```

Comment:
nit: `clusterManager.isEmpty` is cheaper than `runner != null`

```scala
clusterManager.isEmpty && runner != null
```

and make `runner` lazy

Reply:
thanks, changed to var